### DISCUSSION TITLE
Limiting previous token

### DISCUSSION THREAD

**Discussion by amnike** [openai/whisper#851] [Q&A] [answered]
Every time a new result is created it captures the previous result and as this goes on it gets to the point it's capturing the previous 50 results then the processing time increases tremendously lol. 

I get that it's done like this for accuracy but, do you have a way or can point me in the direction of reducing the previous tokens or results captured?

I'd really appreciate it

**Comment by jongwook** [comment, marked as answer]
You could truncate the prefix tokens by adding something like

```
decode_options["prompt"] = decode_options["prompt"][-10:]
```

after:

https://github.com/openai/whisper/blob/0f39c89d9212e4d0c64b915cf7ba3c1f0b59fecc/whisper/transcribe.py#L180

That said, prompt tokens affect the inference time only during the first forward pass through the decoder, which is usually not very significant compared to the total autoregressive decoding time which usually involves tens or hundreds of forward passes through the decoder.

**Reply by amnike** [reply]
Thanks a lot! Any tips on reducing that super lag 15 minutes into a "real-time" transcription?

**Reply by jongwook** [reply]
In case you're referring to #608, I can't comment because I wasn't involved. This repo currently does not support real-time transcription, except an idea I suggested in https://github.com/openai/whisper/discussions/117#discussioncomment-3727051. In any case, you'd need a fast GPU to run the transcription faster than the audio duration.
